client/my-sites/domains/domain-management/edit/transfer.jsx

```diff
@@ -62,6 +62,22 @@ class Transfer extends React.PureComponent {
 				);
 			}
 		} else {
+			transferNotice = (
+				<Notice status={ 'is-error' } showDismiss={ false }>
+					{ translate( 'The transfer has failed. {{a}}Learn more{{/a}}.', {
```

Comment:
Yup, this looks good - we should not land in this state in the normal flow. IIRC, it's only when something unexpected happens at the registrar side and we can't recover.